### PR TITLE
Add summary view

### PR DESCRIPTION
**Pull request by DiegoManzanarezx**
The purpose if this Pull Request is to:

- Add the Summary View on test-runner-frontend with mock data.

### CODE REVIEW

File: test-runner-frontend/src/index.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+const root = ReactDOM.createRoot(document.getElementById('root'));
+root.render(
+ /* <React.StrictMode>
+    <App />
+ </React.StrictMode>
+ */
+<React.StrictMode>
+  <BrowserRouter>
+    <Routes>
+      <Route path="/graphql-java-test-runner" element={<Dashboard/>}/>
```
Comment (DanElliottPalmer) [inline test-runner-frontend/src/index.js:24]: does this mean the dashboard url will be https://adarsh-jaiswal.github.io/graphql-java-test-runner? do you know if we need to look at using the basename prop?

File: test-runner-frontend/package.json [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  "dependencies": {
+    "@emotion/react": "^11.9.3",
+    "@emotion/styled": "^11.9.3",
+    "@mui/icons-material": "^5.8.4",
+    "@mui/material": "^5.8.7",
+    "@mui/x-data-grid": "^5.12.3",
+    "@ramonak/react-progress-bar": "^5.0.1",
+    "@testing-library/jest-dom": "^5.16.4",
+    "@testing-library/react": "^13.3.0",
+    "@testing-library/user-event": "^13.5.0",
```
Comment (DanElliottPalmer) [inline test-runner-frontend/package.json:15]: it looks like we can move some of those dependencies into `devDependencies`

File: test-runner-frontend/src/Components/DropdownMenuAllTestRuns.js [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+import { styled, alpha } from '@mui/material/styles';
+import Button from '@mui/material/Button';
+import Menu from '@mui/material/Menu';
+import MenuItem from '@mui/material/MenuItem';
+import EditIcon from '@mui/icons-material/Edit';
+import Divider from '@mui/material/Divider';
+import ArchiveIcon from '@mui/icons-material/Archive';
+import FileCopyIcon from '@mui/icons-material/FileCopy';
+import MoreHorizIcon from '@mui/icons-material/MoreHoriz';
+import KeyboardArrowDownIcon from '@mui/icons-material/KeyboardArrowDown';
```
Comment (DanElliottPalmer) [inline test-runner-frontend/src/Components/DropdownMenuAllTestRuns.js:11]: can we please remove any imports we're not using?

File: test-runner-frontend/src/Components/DropdownMenuAllTestRuns.js [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+    transformOrigin={{
+      vertical: 'top',
+      horizontal: 'right',
+    }}
+    {...props}
+  />
+))(({ theme }) => ({
+  '& .MuiPaper-root': {
+    borderRadius: 6,
+    marginTop: theme.spacing(1),
```
Comment (DanElliottPalmer) [inline test-runner-frontend/src/Components/DropdownMenuAllTestRuns.js:29]: 🙏 for using built in spacing utils

File: test-runner-frontend/src/Components/DropdownMenuAllTestRuns.js [hunk 5]
```diff
@@ -0,0 +1,108 @@
+import * as React from 'react';
+import { styled, alpha } from '@mui/material/styles';
+import Button from '@mui/material/Button';
+import Menu from '@mui/material/Menu';
```
Comment (DanElliottPalmer) [inline test-runner-frontend/src/Components/DropdownMenuAllTestRuns.js:4]: looking at the figma designs, i think we actually want to be using a select component - https://mui.com/material-ui/react-select/. swapping to this should also help with us not needing to maintain whether it is open or not

File: test-runner-frontend/src/Components/DropdownMenuAllTestRuns.js [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+        backgroundColor: alpha(
+          theme.palette.primary.main,
+          theme.palette.action.selectedOpacity,
+        ),
+      },
+    },
+  },
+}));
+
+export default function DropdownMenuAllTestRuns({childToParent}) {
```
Comment (DanElliottPalmer) [inline test-runner-frontend/src/Components/DropdownMenuAllTestRuns.js:54]: we should rename this to be `onChange`. this component replicates a select element (https://developer.mozilla.org/en-US/docs/Web/HTML/Element/select) and as we're wanting to let the parent know when its value has changed, it mimics the change event (https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/change_event). so we should take naming inspiration from web apis.

File: test-runner-frontend/src/Components/TestRunsTable.js [hunk 7]
```diff
@@ -0,0 +1,241 @@
...
+  },
+  {
+    id: 'density',
+    label: 'Density',
+    minWidth: 170,
+    align: 'right',
+    format: (value) => value.toFixed(2),
+  },
+];
+*/
```
Comment (DanElliottPalmer) [inline test-runner-frontend/src/Components/TestRunsTable.js:44]: can we remove this comment or is it still needed?

File: test-runner-frontend/src/Components/TestRunsTable.js [hunk 8]
```diff
@@ -0,0 +1,241 @@
...
+      minWidth: 300,
+    },
+  ];
+
+function createData(name, code, population, size) {
+  const density = population / size;
+  return { name, code, population, size, density };
+}
+
+export default function TestRunsTable({childToParentCheckBox, checkBoxState, testResults, rowsF, sortBenchmarks, sortDate}) {
```
Comment (DanElliottPalmer) [inline test-runner-frontend/src/Components/TestRunsTable.js:84]: its a little unclear with what some of these props are. this page actually has good recommendations for prop namings that are very common for react users - https://dlinau.wordpress.com/2016/02/22/how-to-name-props-for-react-components/. can we look at renaming these variables to fit this pattern?

File: test-runner-frontend/src/Components/TestRunsTable.js [hunk 9]
```diff
@@ -0,0 +1,241 @@
...
+                  <TableRow hover role="checkbox" tabIndex={-1} key={row.id} sx={{maxHeight: "70px", height: "70px"}}>
+                     
+                    
+                    {columns.map((column) => {
+                      const value = row[column.id];
+                      return (
+                        
+                        <TableCell key={column.id} align={column.align}>
+                            
+                        {
```
Comment (DanElliottPalmer) [inline test-runner-frontend/src/Components/TestRunsTable.js:168]: ternary statements can sometimes be a little hard to read. especially when they're big like this. can we please look at splitting these into small components? doing something like this should make it a little easier to read:
```jsx
{columns.map((column) => {
  const value = row[column.id];

	let cell = null;
	if(column.id === "improvedVsRegressed"){
		cell = <ComparisonCell />
	} else if(column.id !== "id"){
		cell = value
	// etc
	}

  return (
	  <TableCell key={column.id} align={column.align}>
	    {cell}
	  </TableCell>
  )
})}
```

**Review by DanElliottPalmer — COMMENTED** [review]
looks great! a few other things we should do before we merge this in:

- remove any commented out code we don't need anymore
- can we please add [prettier](https://prettier.io/) to this project? some of the indentation make it a little hard to read the code. however, we can automate a script that will run over the files and just tidy it. i would recommend that we add a new npm script that runs prettier over js files in src/.
- i noticed there are some eslint warnings when running it locally. even though theyre warnings, it would be good to correct them.

File: test-runner-frontend/src/index.js [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+  Routes,
+} from "react-router-dom";
+import { Dashboard } from './Views/Dashboard';
+
+const root = ReactDOM.createRoot(document.getElementById('root'));
+root.render(
+ /* <React.StrictMode>
+    <App />
+ </React.StrictMode>
+ */
```
[the remark this answers is not in the document]
Comment (DiegoManzanarezx) [reply test-runner-frontend/src/index.js:20]: Done.

File: test-runner-frontend/src/index.js (hunk 1 above)
Comment (DiegoManzanarezx) [reply test-runner-frontend/src/index.js:24]: Yes, that will be the url, and I've already added the basename prop.

File: test-runner-frontend/package.json (hunk 2 above)
Comment (DiegoManzanarezx) [reply test-runner-frontend/package.json:15]: Done, also I deleted some dependencies we are not using.

File: test-runner-frontend/src/Components/TestRunsTable.js (hunk 7 above)
Comment (DiegoManzanarezx) [reply test-runner-frontend/src/Components/TestRunsTable.js:44]: It is not needed. I removed it already.

File: test-runner-frontend/src/Components/DropdownMenuAllTestRuns.js [hunk 11]
```diff
@@ -0,0 +1,108 @@
...
+  };
+
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
+
+  return (
+    <div>
+      <Button
+        id="demo-customized-button"
```
[the remark this answers is not in the document]
Comment (DiegoManzanarezx) [reply test-runner-frontend/src/Components/DropdownMenuAllTestRuns.js:81]: No, I removed them.

File: test-runner-frontend/src/Components/TestRunsTable.js (hunk 9 above)
Comment (DiegoManzanarezx) [reply test-runner-frontend/src/Components/TestRunsTable.js:168]: I didn't know we could use if else conditions, that's super cool! I lalready made the change, thank you!

File: test-runner-frontend/src/Components/DropdownMenuAllTestRuns.js [hunk 12]
```diff
@@ -0,0 +1,108 @@
...
+    anchorOrigin={{
+      vertical: 'bottom',
+      horizontal: 'right',
+    }}
+    transformOrigin={{
+      vertical: 'top',
+      horizontal: 'right',
+    }}
+    {...props}
+  />
```
[the remark this answers is not in the document]
Comment (DiegoManzanarezx) [reply test-runner-frontend/src/Components/DropdownMenuAllTestRuns.js:25]: This component will no longer be used, instead we'll use Select MUI as you recommended.

File: test-runner-frontend/src/Views/Dashboard.js [hunk 13]
```diff
@@ -0,0 +1,550 @@
...
+              right: "2%",
+              size: "small"
+            }}
+              variant="outlined"
+              disabled={isCheckBoxActive && checkBoxSelection.length !== 2 ? true : false}
+              onClick={() => manageCompareAction()}
+            >
+              Compare
+            </Button>
+          </Stack>
```
Comment (DanElliottPalmer) [inline test-runner-frontend/src/Views/Dashboard.js:473]: as we've not fully implemented this yet and will be later ticket, can we please comment this block. also i know ive just told you to remove all commented out code but we'll want to likely keep it for when we do that ticket. unless you want to remove it for the time being.

**Review by DanElliottPalmer — COMMENTED** [review]
yesssssss. only one comment around the compare button and i noticed theres still some lint warnings. can we please address them? once both of those are good, ill hit the approve button 👍

File: test-runner-frontend/src/Views/Dashboard.js (hunk 13 above)
Comment (DiegoManzanarezx) [reply test-runner-frontend/src/Views/Dashboard.js:473]: Yes, I commented that block as well as code that will be used for next ticket in order to get rid of the lint warnings.

**Review by DanElliottPalmer — APPROVED** [review]

**Pull request merged**